### DISCUSSION TITLE
Generalization as a function of zoom level - polygons

### DISCUSSION THREAD

**Discussion by ymoisan** [developmentseed/lonboard#651] [General]
Hi.  I downloaded building footprint data (polygons) from 'https://ftp.maps.canada.ca/pub/nrcan_rncan/extraction/auto_building/shp/, combined all shapefiles there into one big geodataframe and saved the result as geoparquet with `geometry_encoding='geoarrow'`.  This gives a table of 11M lines.

I tried opening the parquet file with `geoarrow.rust.core` `read_parquet` and use it in a PolygonLayer : 

'geoarrow.rust.core._rust.GeoTable' object [table_parquet_geoarrow_rust] has no attribute 'select'

So I ended up opening the parquet file with `pyarrow.parquet` `read_table` and now I can get a PolygonLayer.  I have a GPU-less computer and got to see parts of the map but display is awfully slow.  I would think 11M polygons is just too much to draw.

I'm trying to mimick [this map](https://app.geo.ca/map?rvKey=7a5cda52-c7df-427f-9ced-26f19a8a64d6), which shows rough perimeter of areas with data and more and more detail as one zooms in, only with the data in lonboard (and eventually Polars/DuckDB for analysis).  Have there been discussions on ways to adapt the number of vertices in polygons as a function of zoom level e.g. take a bbox or some rough perimeter of individual polygons ?  Or get some rough perimeter of groups of polygons, a kind of "geometry aggregation", as a function of zoom level ?

**Comment by kylebarron** [comment]
> 'geoarrow.rust.core._rust.GeoTable' object [table_parquet_geoarrow_rust] has no attribute 'select'

Regardless, you should ensure you're using the latest version of geoarrow-rust.

At this point, geometry simplification is not in scope of Lonboard, and you should do any geometry simplification you need before passing the data to Lonboard

**Comment by ymoisan** [comment]
> Regardless, you should ensure you're using the latest version of geoarrow-rust

Sorry for that.  geoarrow.rust.io 0.3.0 read_parquet works fine.

**Reply by kylebarron** [reply]
geoarrow-rust-compute has some linestring simplification algorithms you can try: https://geoarrow.org/geoarrow-rs/python/latest/api/compute/functions/#geoarrow.rust.compute.simplify. These won't maintain a polygonal coverage, if you need that
